refinery/analysis_manager/utils.py

```diff
@@ -312,43 +181,10 @@ def get_solr_results(query, facets=False, jsonp=False, annotation=False,
     return results
 
 
-def match_nodesets(ns1, ns2, diff_f, all_f, rel_type=None):
-    """Helper function for matching 2 nodesets solr results"""
-    logger.debug("analysis_manager.views match_nodesets called")
-    ret_info = {}
-    ret_info['total'] = str(len(ns1) + len(ns2))
-    ret_info['node1_count'] = str(len(ns1))
-    ret_info['node2_count'] = str(len(ns2))
-
-    best_list = []
-    template = {'uuid_1': '', 'uuid_2': '', 'frac': 0.0, 'same': 0, 'diff': 0,
-                'tot': 0, 'sel_tot': 0, 'sel': 0, 'sel_frac': 0.0}
-    i = 0
-    for node1 in ns1:
-        best_node = template.copy()
-        j = 0
-        for node2 in ns2:
-            if node1['uuid'] != node2['uuid']:
-                temp_node = template.copy()
-                temp_node['uuid_1'] = node1['uuid']
-                temp_node['uuid_2'] = node2['uuid']
-                # counts differences for list of fields
-                for df in diff_f:
-                    # if the given column matches between Nodesets
-                    if node1[df] == node2[df]:
-                        best_node = temp_node
-            j += 1
-        best_list.append(best_node)
-        i += 1
-    # matches
-    ret_info['matches'] = str(len(best_list))
-    return best_list, ret_info
-
-
 def validate_analysis_config(analysis_config):
     """
-    Validate incoming Tool Launch Configurations
-    :param tool_launch_config: json data containing a ToolLaunchConfiguration
+    Validate incoming Analysis Configurations
+    :param analysis_config: json data containing an Analysis configuration
     """
     with open(
```

Comment:
All the JsonSchema validations are within the scopes of atomic transactions so I want any exceptions to propagate up. I'll make a note.

Reply:
OK, a comment at the top of this function would be helpful. Btw, it looks like AnalysisConfig.json is essentially a configuration file, so I would move the reading operation out into base settings (similar to tutorial_steps.json, etc). This would make validate_analysis_config() cleaner and more testable. Also, should this file be named using snake case or camel case?